Feature/chat assistant ollama

Moderndo opened the pull request:
Chat Assistant: FAQ fallback, Support package cleanup, UI consistency”
Changes in this PR
- Fixed build conflicts in XML (colors.xml, arrays.xml, strings.xml).
- Added FAQ fallback intent in ChatAssistantActivity for common user queries.
- Cleaned up Support package (moved classes into correct package, removed stray files).
- Unified chat UI with app theme (consistent colors, text styles).
- Verified app builds & runs: chat, send/receive, history button all work.

Moderndo committed:
Merge upstream/dev into main

Moderndo committed:
docs: add emulator screenshots, notes, and README run instructions

- Added screenshots of emulator running the app
- Created notes file for setup and testing documentation
- Updated README with server start instructions
- Committed changes to ChatAdapter.java and other updated files

Moderndo committed:
Remove misc.xml conflict from merge with dev

Moderndo committed:
Remove merge conflict markers from SettingsActivity.java

Moderndo committed:
Update SharedActivity.java

Moderndo committed:
Update strings.xml

Moderndo committed:
Clean colors.xml (remove duplicates, fix names)

Moderndo committed:
Fix arrays.xml to be valid

Moderndo committed:
Fix strings.xml (merge into single resources block, remove duplicates)

Moderndo committed:
Fix ChatAssistantActivity and strings.xml (resolve conflicts, rename your_request)

Lillian-ran reviewed (changes requested):
Thanks for the fixes—build issues are resolved on my side too. Great work cleaning up the conflicts and duplicates. I have a few follow‑ups to align this screen with the rest of the app and improve the chat experience:

Thanks for the fixes—build issues are resolved on my side too.  Great work cleaning up the conflicts and duplicates.  I have a few follow-ups to align this screen with the rest of the app and improve the codebase overall:

1) ChatAssistantActivity – stability & structure

Please ensure the final version keeps one clear init path (findViewById → setup RecyclerView → seed welcome → init OllamaClient → wire sendButton).

Avoid wrapping all of onCreate() in a try/catch that just calls finish() on error—this can hide real issues and cause blank screens.

Make sure message types are consistent (ChatMessage. BOT / ChatMessage. USER) and always scroll after adding a message.

2) Theme & colors – keep consistent with the app

Right now the chat screen colors feel different from the rest of the app.

Please revert to the default app palette (white background, deep-blue text) and avoid introducing page-specific overrides in layouts/styles.

Use centralized values from colors.xml instead of hard-coded hex.

3) Chat intelligence – handle general user questions

The bot currently fails on basic questions like “What is this app?”.

Please add a fallback FAQ intent layer before calling the LLM, with common Qs like:

“what is this app” → “This is a Smishing Detection app that helps you identify and report SMS phishing…”

“how do I report” → “Use the Community Report tab to submit suspicious messages…”

Ensure there’s always a meaningful default reply if no intent matches.

4) Code structure – Support package

There is now both a Support folder under src/main/java/ (outside our app package) and also a com/example/smishingdetectionapp/Support package.

The top-level Support folder looks misplaced—SupportActivity.java should live under com.example.smishingdetectionapp.Support.

There is also a stray Support file (green “+” in the screenshot) that seems to be an accidental empty Java file.  This can be safely deleted.

Please move/merge everything so that all support-related classes live under app/src/main/java/com/example/smishingdetectionapp/Support/.  That keeps package structure clean and consistent.

Once these are addressed, I’m happy to re‑review. Thanks again!

Moderndo committed:
feat(chat): FAQ fallback, Support package cleanup, consistent UI updates

Moderndo commented:
Hi Lillian, I’ve addressed your previous feedback:
	•	Removed duplicate/invalid XML entries.
	•	Added FAQ fallback responses in ChatAssistantActivity.
	•	Cleaned up Support package.
	•	Verified UI matches theme + build works.
 All features are working and its now good to merge, Please re-review thanks for your guidance
<img width="335" height="764" alt="Screenshot 2025-08-27 163309" src="https://github.com/user-attachments/assets/480529fc-8465-4d9a-99fc-efac1d532a0b" />
<img width="342" height="771" alt="image" src="https://github.com/user-attachments/assets/cfe35da6-a274-4344-84bb-57e785fe08c4" />
Youtube video lInk:
https://youtube.com/shorts/EibFnFgNnoU?feature=share
https://youtube.com/shorts/k_1WXKLUIuA?feature=share

Moderndo committed:
Merge branch 'dev' into feature/chat-assistant-ollama-

Moderndo committed:
Update bubble_bot.xml

Moderndo committed:
Update bubble_user.xml

Lillian-ran reviewed (changes requested):
Thanks for the updates!
However, I noticed some things when reviewing this PR:

The project still fails to build on my side due to a :app:mergeDebugResources error. This seems to come from recent changes in the resource files (values). Please re-check and fix this so the branch can build successfully.

Regarding the design style of the chat page: I had previously requested that the colors follow the app’s existing theme (white background, deep-blue text). In this PR, I still see the earlier version with the custom colors. It looks like there may have been an issue when committing or creating the PR, and the latest changes weren’t included.

Could you please review these points and update the PR accordingly? Thanks!
<img width="859" height="933" alt="image" src="https://github.com/user-attachments/assets/4f45b485-51e5-46b2-8f3e-486103f7a815" />
